axelor-account/src/main/resources/views/Invoice.xml

```diff
@@ -2885,4 +2886,10 @@
     <action name="close"/>
   </action-group>
 
+  <action-method
+    name="action-invoice-method-check-invoice-lines-not-only-non-deductible-taxes">
+    <call class="com.axelor.apps.account.web.InvoiceController"
+      method="checkTaxesNotOnlyNonDeductibleTaxes"/>
+  </action-method>
```

Comment:
this check should be done when we select the taxes on the InvoiceLine, not when we validate the invoice.
And also please add this check when we add taxes everywhere there is a M2M taxes